test/e2e/cloud/gcp/common/upgrade_mechanics.go

```diff
@@ -254,15 +254,16 @@ func checkNodesVersions(ctx context.Context, cs clientset.Interface, want string
 		// We do prefix trimming and then matching because:
 		// want   looks like:  0.19.3-815-g50e67d4
 		// kv/kvp look  like: v0.19.3-815-g50e67d4034e858-dirty
-		kv, kpv := strings.TrimPrefix(n.Status.NodeInfo.KubeletVersion, "v"),
-			strings.TrimPrefix(n.Status.NodeInfo.KubeProxyVersion, "v")
+		kv := strings.TrimPrefix(n.Status.NodeInfo.KubeletVersion, "v")
+		kpv := n.Status.NodeInfo.KubeProxyVersion //nolint:staticcheck // Keep testing deprecated KubeProxyVersion field until it's being removed
 		if !strings.HasPrefix(kv, want) {
 			return fmt.Errorf("node %s had kubelet version %s which does not start with %s",
 				n.ObjectMeta.Name, kv, want)
 		}
-		if !strings.HasPrefix(kpv, want) {
-			return fmt.Errorf("node %s had kube-proxy version %s which does not start with %s",
-				n.ObjectMeta.Name, kpv, want)
+
+		if len(kpv) != 0 {
+			return fmt.Errorf("node %s had kube-proxy version %s which should be empty",
+				n.ObjectMeta.Name, kpv)
```

Comment:
The test shouldn't just assume that the feature gate is set to its default value; you can run the tests with various combinations of feature gates.

You can either check the feature gate here and then check the appropriate result depending on that, or else ignore the feature gate and make the test accept either value ("empty" or "matches kubelet version").

(Yeah, this should have been changed in alpha.)

Reply:
> You can either check the feature gate here and then check the appropriate result depending on that, or else ignore the feature gate and make the test accept either value ("empty" or "matches kubelet version").

We can't seem to detect whether the feature gate is enable in the e2e test, so I choose accept either value.